apps/src/lib/node/ledger/mod.rs

```diff
@@ -609,42 +610,58 @@ fn start_tendermint(
         })
 }
 
-/// Launches a new task managing a `geth` process into the asynchronous
-/// runtime, and returns its [`task::JoinHandle`].
-///
-/// An oracle is also returned, along with its associated channel,
-/// for receiving Ethereum events from `geth`.
-async fn maybe_start_ethereum_oracle(
+/// Potentially starts an Ethereum event oracle.
+fn maybe_start_ethereum_oracle(
     config: &config::Ledger,
     abort_sender: oneshot::Sender<()>,
-) -> (Option<mpsc::Receiver<EthereumEvent>>, task::JoinHandle<()>) {
+) -> (
+    Option<mpsc::Receiver<EthereumEvent>>,
```

Comment:
I noticed that  we don't have a return pattern of the following `(None, Some(_), handle)`. So this return is not well typed. We should probably return an enum instead with names reflecting the dfiferent modes

Reply:
Yeah an enum is way better here